tests/test_yaml.py

```diff
@@ -29,3 +30,14 @@ def test_file_read(datadir):
     file_handle = open(file_to_load, 'r')
     actual = parse(file_handle.read())
     assert actual == expected
+
+
+def test_write(datadir, tmp_path):
```

Comment:
I think this test should be split in two - one success case and one fail case.  In each one, you can diff either the way you have or by comparing the key load from each of the files you just wrote.  I would suggest the comparison be separate from the writing though, so you are testing the contents of the file instead of whether the writing failed (at least, I _think_ that is what you mean to test here?).  However, now that I say that, what happens if a file fails to write?  I don't see any error to handle that in the code (say, unwritable directory or something).

Reply:
Separating success and failure tests sounds like a good idea.

Reply:
I added good write and bad write tests and then changed this test to a round trip test. Hopefully that helps to clarify. Note the round trip test is about formatting.